### PR TITLE
Implement Job Log Updates

### PR DESCRIPTION
**Pull request by irfanshar**
- Replace command level tree item with <object> as the label and <source> as the description
- The above should be colored red if the command failed (or at least the icon for it will be a red x or green check)
- Add the command at the same level of the messages
- Add action to toggle failures
- Add action to filter in order to view above a severity level
- Add action to view output for individual objects

### CODE REVIEW

File: src/views/jobLog/log.ts [hunk 1]
```diff
@@ -14,24 +14,37 @@ import { ContextValue } from "../../ibmiProjectExplorer";
...
 
     const jobLogInfo = this.jobLogInfo;
 
-    items.push(...jobLogInfo.commands?.map(
-      command => new Command(this.workspaceFolder, command)
-    ));
-
+    if (this.showFailedCommands) {
+      items.push(...jobLogInfo.commands?.filter(
+        command => command.failed === this.showFailedCommands
```
Comment (edmundreinhardt) [inline src/views/jobLog/log.ts:39]: you already know the `showFailedCommands` is `true`
so why not just have `command.failed`

**Review by edmundreinhardt — CHANGES_REQUESTED** [review]

**Review by edmundreinhardt — CHANGES_REQUESTED** [review]
Behaviour changes
1) the icon for showing only failing commands, it should be a red x of some sort, it should appear differently whether it si toggled on or off
2) the filter by severity should be shown for the same scope, i.e. for all the entire build.
i.e. it will apply to all the commands for the current build.  
3) The copy command should copy the command (which is now a child) rather than just the name of the targer object

File: src/views/jobLog/index.ts [hunk 2]
```diff
@@ -88,15 +88,23 @@ export default class JobLog implements TreeDataProvider<ProjectExplorerTreeItem>
...
           element.toggleShowFailed();
           this.refresh(element);
         }
       }),
-      commands.registerCommand(`vscode-ibmi-projectexplorer.jobLog.filterMessageSeverity`, async (element: Command) => {
+      commands.registerCommand(`vscode-ibmi-projectexplorer.jobLog.showAllJobs`, async (element: Log) => {
+        const iProject = ProjectManager.get(element.workspaceFolder);
+
+        if (iProject) {
+          element.toggleShowFailed();
```
Comment (edmundreinhardt) [inline src/views/jobLog/index.ts:103]: hopefully these never get out of sync.  It would be safer to set the desired value.  Although i see this works.

**Review by edmundreinhardt — APPROVED** [review]

**Comment by edmundreinhardt** [comment]
Hmm, I am not seeing the action toggle

**Comment by irfanshar** [comment]
> Hmm, I am not seeing the action toggle

The two commands that use toggle are ```vscode-ibmi-projectexplorer.jobLog.showOnlyFailedJobs``` and ```vscode-ibmi-projectexplorer.jobLog.showAllJobs```.  I felt these were more descriptive names rather than two different "toggle" commands.

**Comment by edmundreinhardt** [comment]
when I tested, and clicked the button, it did not toggle

**Comment by irfanshar** [comment]
> when I tested, and clicked the button, it did not toggle

Hmm, odd it is toggling on my end. Can you attach a copy of the joblog.json you're using to test?

**Pull request merged**